lib/grizzly/zwave_firmware.ex

```diff
@@ -0,0 +1,155 @@
+defmodule Grizzly.ZwaveFirmware do
+  @moduledoc """
+  Controller ZWave firmware update support
+  """
+  require Logger
+
+  @doc """
+  Update the ZWave chip firmware if zw_programmer is present and a
+  valid GBL file is available that would up the firmware version.
+  """
+  @spec maybe_run_zwave_firmware_update(Grizzly.Options.t()) :: :ok
+  def maybe_run_zwave_firmware_update(options) do
+    with :ok <- check_firmware_dir(options),
+         :ok <- check_zwave_programmer_exists(options),
+         {:ok, gbl_file} <- get_firmware_file(options),
+         :ok <- maybe_upload_zwave_firmware_update(gbl_file, options) do
+      :ok
+    else
+      {:error, :no_firmware_dir} ->
```

Comment:
It doesn't look like there's any handling of the error atom except to select something to write to a log. How about getting rid of all of the error atoms and return the message? Like this:

```suggestion
      {:error, message} ->
        Logger.info("[Grizzly]: firmware updated failed due to #{message}"
```